File: backend/app/models/politics.py
from sqlalchemy import Column, Integer, String, Text, Date, DateTime, ForeignKey
from sqlalchemy.sql import func
from ..database import Base


class PoliticsRecitation(Base):
    __tablename__ = "politics_recitations"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    title = Column(String(200), nullable=False)
    category = Column(String(50), nullable=False, default="马原")
    content = Column(Text, nullable=False)
    image_path = Column(String(500), nullable=True)
    mastery_level = Column(String(20), nullable=False, default="生疏")
    review_count = Column(Integer, nullable=False, default=0)
    next_review_date = Column(Date, nullable=True)
    last_review_date = Column(DateTime(timezone=True), nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())


class RecitationReminder(Base):
    __tablename__ = "recitation_reminders"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    reminder_type = Column(String(20), nullable=False, default="单词")
    reminder_time = Column(String(10), nullable=False, default="08:00")
    frequency = Column(String(20), nullable=False, default="每天")
    enabled = Column(Integer, nullable=False, default=1)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

File: frontend/src/api/politics.ts
import axios from './axios'

export interface PoliticsRecitation {
  id: number
  user_id: number
  title: string
  category: string
  content: string
  image_path: string | null
  mastery_level: string
  review_count: number
  next_review_date: string | null
  last_review_date: string | null
  created_at: string
  updated_at: string | null
}

export interface RecitationReminder {
  id: number
  user_id: number
  reminder_type: string
  reminder_time: string
  frequency: string
  enabled: number
  created_at: string
  updated_at: string | null
}

export interface RecognizeResponse {
  content: string
  confidence: number
}

export interface UploadResponse {
  image_path: string
  image_url: string
}

export async function getRecitations(params?: {
  category?: string
  mastery_level?: string
}): Promise<PoliticsRecitation[]> {
  return await axios.get('/api/politics', { params })
}

export async function getRecitation(id: number): Promise<PoliticsRecitation> {
  return await axios.get(`/api/politics/${id}`)
}

export async function createRecitation(data: {
  title: string
  category?: string
  content: string
  image_path?: string
}): Promise<PoliticsRecitation> {
  return await axios.post('/api/politics', data)
}

export async function updateRecitation(id: number, data: Partial<{
  title: string
  category: string
  content: string
  image_path: string
  mastery_level: string
}>): Promise<PoliticsRecitation> {
  return await axios.put(`/api/politics/${id}`, data)
}

export async function deleteRecitation(id: number): Promise<void> {
  await axios.delete(`/api/politics/${id}`)
}

export async function reviewRecitation(id: number, result: string): Promise<PoliticsRecitation> {
  return await axios.post(`/api/politics/${id}/review`, { result })
}

export async function uploadPoliticsImage(file: File): Promise<UploadResponse> {
  const formData = new FormData()
  formData.append('file', file)
  return await axios.post('/api/politics/upload', formData, {
    headers: { 'Content-Type': 'multipart/form-data' }
  })
}

export async function recognizePoliticsImage(imagePath: string): Promise<RecognizeResponse> {
  return await axios.post('/api/politics/recognize', { image_path: imagePath })
}

export async function getReminders(reminderType?: string): Promise<RecitationReminder[]> {
  return await axios.get('/api/politics/reminders', { params: { reminder_type: reminderType } })
}

export async function createReminder(data: {
  reminder_type: string
  reminder_time: string
  frequency: string
}): Promise<RecitationReminder> {
  return await axios.post('/api/politics/reminders', data)
}

export async function updateReminder(id: number, data: Partial<{
  reminder_time: string
  frequency: string
  enabled: number
}>): Promise<RecitationReminder> {
  return await axios.put(`/api/politics/reminders/${id}`, data)
}

export async function deleteReminder(id: number): Promise<void> {
  await axios.delete(`/api/politics/reminders/${id}`)
}
